docs/contributing/pull-requests/branching.md

```diff
@@ -19,8 +19,9 @@ parts, separated by a slash:
 
 - The team name or abbreviation (e.g. `Vault`), and
 - The Jira issue tag (e.g. `PM-1234`)
+- A short description of the work being done (e.g. `update-csp-hashes`)
 
-In this example, the full branch name would be `Vault/PM-1234`.
+In this example, the full branch name would be `Vault/PM-1234/update-csp-hashes`.
```

Comment:
💭 What's the benefit of adding this vs. just using a task ID? Our PR template now expects a link to the task and it'll be linked inside Jira. Might be better to _remove_ the task ID here?

Reply:
I wasn't going to, but I'll bite: why not just use the auto-generated jira branch name? Then we just don't have to worry about it. Not sure if I've missed the boat on this discussion, but this seems tedious to follow and enforce.

Reply:
@withinfocus the task id in the branch is the means by we can guarantee jira detects a branch belongs to a given ticket. There are other ways, but they don't generate easily searchable branch trees. I also want to be able to understand the gist of what a branch is doing by looking at the name, not go into another tool/web page to determine that. Finally, the description is last to allow auto-completing on the unique ticket.

@eliykat Jira creates branches something along the lines of "board_ticket_description", but the visualization range given to tickets is FAR more than is typical for a branch. Underscores are also not interpreted as separators for branch groups, and it produces the same branch name for each ticket, but we want to allow multiple per ticket if that's appropriate. In short: I just think Jira does a bad job at this.

Reply:
Using the team identifier first allows you to filter out 90% of the branches if your tool supports it. 
<img width="334" alt="image" src="https://github.com/bitwarden/contributing-docs/assets/137855/e5fceba6-795b-49d7-99bf-86747600b9a2">

Reply:
What's the reason for browsing branches? I guess I don't understand the basis.

Reply:
I regularly have a bunch of PR reviews in flight, I use branch browsing to keep by local repos organized.